web/react/utils/utils.jsx

```diff
@@ -385,112 +385,262 @@ module.exports.searchForTerm = function(term) {
     });
 }
 
-var oldExplicitMentionRegex = /(?:<mention>)([\s\S]*?)(?:<\/mention>)/g;
-var puncStartRegex = /^((?![@#])\W)+/g;
-var puncEndRegex = /(\W)+$/g;
+/* Options:
+      - disable: Parses out *'s and other format specifiers in the text, but doesn't convert to html
+*/
+module.exports.customMarkedRenderer = function(options) {
+    var customTextRenderer = new formatText.Renderer();
+    if (options && options.disable) {
+        customTextRenderer.paragraph = function(text) {
+            return text + ' ';
+        };
+        customTextRenderer.strong = function(text) {
+            return text;
+        };
+        customTextRenderer.em = function(text) {
+            return text;
+        };
+        customTextRenderer.codespan = function(code) {
+            return code;
+        };
+        customTextRenderer.link = function(href) {
+            return href;
+        };
+        customTextRenderer.image = function(href) {
+            return href;
+        };
+    } else {
+        customTextRenderer.link = function(href) {
+            return href;
+        };
+        customTextRenderer.image = function(href) {
+            return href;
+        };
+    }
+    return customTextRenderer;
+};
+
+var puncStartRegex = /^((?![@#])[^A-Za-z0-9_<>])+/g;
+var puncEndRegex = /([^A-Za-z0-9_<>])+$/g;
+var startTagRegex = /(<\s*\w.*?>)+/g;
+var endTagRegex = /(<\s*\/\s*\w\s*.*?>|<\s*br\s*>)+/g;
+
+module.exports.textToJsx = function(textToChange, options) {
```

Comment:
Yeah @hmhealey and I were talking about that with regards to the original implementation of this in #327 and how it was becoming a nightmare to work with.